graylog2-web-interface/src/components/indices/rotation/TimeBasedSizeOptimizingStrategyConfiguration.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+
+import useMaxIndexRotationLimit from 'hooks/useMaxIndexRotationLimit';
+import RangeInput from 'components/common/RangeInput';
+
+export type TimeBasedSizeOptimizingStrategyConfig = {
+  index_lifetime_max: string,
+  index_lifetime_min: string,
+  type: string,
+}
+
+type Props = {
+  config: TimeBasedSizeOptimizingStrategyConfig,
+  updateConfig: (config: Omit<TimeBasedSizeOptimizingStrategyConfig, 'type'>) => void,
+}
+
+export const durationToRoundedDays = (duration: string) => {
+  return Math.round(moment.duration(duration).asDays());
+};
+
+const getRangeInDays = (indexLifeTimeMin = 'PT1H', IndexLifeTimeMax = 'P4D') => {
```

Comment:
If the methods (except for types and constants) are not shared I think they can either be put into the component or if they should really be extracted in their own helper file. Otherwise, I don't see the point of moving them out of the component.

Reply:
I usually move helper functions with no state outside of the component to avoid recreating it on each render.

Reply:
Ah, good point! That makes sense!